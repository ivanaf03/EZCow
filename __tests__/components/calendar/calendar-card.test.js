import React from "react";

import { render } from "@testing-library/react-native";

import CalendarCard from "../../../components/calendar/calendar-card";

describe("CalendarCard", () => {
  it("should render correctly", () => {
    const tree = render(
      <CalendarCard date="2022-01-01" events={[]} isSelected={false} />
    );

    expect(tree).toMatchSnapshot();
  });

  it("should render correctly when selected", () => {
    const tree = render(
      <CalendarCard date="2022-01-01" events={[]} isSelected={true} />
    );

    expect(tree).toMatchSnapshot();
  });

  it("should render correctly with events", () => {
    const tree = render(
      <CalendarCard date="2022-01-01" events={[{ name: "test", description: "test" }]} isSelected={false} />
    );

    expect(tree).toMatchSnapshot();
  });
});