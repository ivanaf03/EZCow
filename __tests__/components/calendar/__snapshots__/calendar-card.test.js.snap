// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarCard should render correctly 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "width": "100%",
      },
      {
        "justifyContent": "flex-start",
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      {
        "borderBottomColor": "#ffffff",
        "borderLeftColor": "#ffffff",
        "borderRightColor": "#ffffff",
        "borderTopColor": "#ffffff",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        {
          "fontWeight": "700",
        },
        {
          "fontFamily": "Nunito-Medium",
        },
      ]
    }
  >
    2022-01-01
  </Text>
</View>
`;

exports[`CalendarCard should render correctly when selected 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "width": "100%",
      },
      {
        "justifyContent": "flex-start",
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "borderBottomWidth": 8,
        "borderLeftWidth": 8,
        "borderRightWidth": 8,
        "borderTopWidth": 8,
      },
      {
        "borderBottomColor": "#6bc0eb",
        "borderLeftColor": "#6bc0eb",
        "borderRightColor": "#6bc0eb",
        "borderTopColor": "#6bc0eb",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        {
          "fontWeight": "700",
        },
        {
          "fontFamily": "Nunito-Medium",
        },
      ]
    }
  >
    2022-01-01
  </Text>
</View>
`;

exports[`CalendarCard should render correctly with events 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "width": "100%",
      },
      {
        "justifyContent": "flex-start",
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      {
        "borderBottomColor": "#ffffff",
        "borderLeftColor": "#ffffff",
        "borderRightColor": "#ffffff",
        "borderTopColor": "#ffffff",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        {
          "fontWeight": "700",
        },
        {
          "fontFamily": "Nunito-Medium",
        },
      ]
    }
  >
    2022-01-01
  </Text>
  <View
    style={
      [
        {
          "paddingBottom": 8,
        },
        {
          "marginBottom": 8,
          "marginTop": 8,
        },
        {
          "flexDirection": "column",
        },
        {
          "justifyContent": "space-between",
        },
        {
          "borderBottomWidth": 2,
        },
        {
          "borderBottomColor": "#6bc0eb",
          "borderLeftColor": "#6bc0eb",
          "borderRightColor": "#6bc0eb",
          "borderTopColor": "#6bc0eb",
        },
        {
          "borderStyle": "dashed",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
          },
          {
            "fontFamily": "Nunito-BoldItalic",
          },
          {
            "textDecorationLine": "underline",
          },
        ]
      }
    >
       - 
      test
    </Text>
    <Text
      style={
        [
          {
            "color": "#ffffff",
          },
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "fontFamily": "Nunito-Medium",
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;
